sympy/solvers/bivariate.py

```diff
@@ -151,7 +154,11 @@ def _lambert(eq, x):
 
     # invert the generator X1 so we have x(u)
     u = Dummy('rhs')
-    xusolns = solve(X1 - u, x)
+    if not domain.is_subset(S.Reals):
+        dmn = False
+    else:
+        dmn = True
+    xusolns = solve(X1 - u, x,dmn=dmn)
```

Comment:
There shouldn't be a `dmn` flag to solve.

Reply:
On adding a random flag (domain -->dnm<for domain consistency>) does the signature of solve change!?
I think the functionality remains the same and I think it's like an internal API...

Also, I think the alternate should be present instead of `solve`{imo that could be `solvify` function } to maintain consistent domain.

Reply:
As of now I not able to find any other means for domain consistency.
I've tried 
1. global variable
2. passing domain to `solve`, `_tsolve`, `_solve`
3. presently, this flag as `dmn`